src/viam/sdk/resource/resource.hpp

```diff
@@ -18,10 +18,9 @@ class Resource {
    public:
     virtual ~Resource();
     explicit Resource(std::string name);
-    static API static_api();
 
     /// @brief Returns the `API` associated with a particular resource.
-    virtual API dynamic_api() const = 0;
+    virtual API api() const = 0;
```

Comment:
[q] I know this isn't `dynamic_api` anymore, but where do we need access to this second "dynamic" `api` method? When do we not know the name of the class whose static `api` method to call?

Reply:
Yeah, good question! It's needed when doing methods on all resources on a `robot` where we don't know what type of `resource` we have in the abstract. So for `generate_status`, `stop_all`, etc. we need a proto `ResourceName` and we build that from the `API`.

Reply:
> we need a proto ResourceName and we build that from the API.

Hmm I see; I suppose I question whether you need a fully qualified `ResourceName` (or just the short version) to run those methods. Seems fine to have both for now, though!